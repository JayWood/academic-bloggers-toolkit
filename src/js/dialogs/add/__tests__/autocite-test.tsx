import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as React from 'react';
import AutoCite from '../autocite';

const getter = jest.fn();

const setup = ({ kind = 'webpage', inputType = 'url' } = {}, pattern?: any) => {
    const component = mount(
        <AutoCite
            kind={kind as any}
            inputType={inputType as any}
            placeholder="Test"
            pattern={pattern}
            getter={getter}
        />,
    );
    return {
        component,
    };
};

describe('<AutoCite />', () => {
    beforeEach(() => jest.resetAllMocks());
    it('should match snapshots', () => {
        let { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();

        ({ component } = setup({ kind: 'book', inputType: 'text' }, '[0-9]+'));
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should handle field change', () => {
        const { component } = setup();
        const query = component.find('#citequery');
        query.simulate('change', { currentTarget: { value: 'hello' } });
    });
    it('should call handleQuery', () => {
        const { component } = setup();
        const instance = (component as any).instance();
        instance.input.validity = {
            valid: true,
        };
        instance.query.set('testing');
        const button = component.find('input[type="button"]');

        button.simulate('click');
        expect(instance.query.get()).toBe('');
        expect(getter).toHaveBeenCalledTimes(1);

        instance.input.validity = {
            valid: false,
        };
        instance.query.set('foo');
        button.simulate('click');

        expect(instance.query.get()).toBe('foo');
        expect(getter).toHaveBeenCalledTimes(1);
    });
    it('should call handleKeyDown', () => {
        const { component } = setup();
        const input = component.find('#citequery');
        const stopPropagation = jest.fn();
        const preventDefault = jest.fn();

        input.simulate('keyDown', { key: 'Enter', stopPropagation, preventDefault });
        input.simulate('keyDown', { key: 'Return', stopPropagation, preventDefault });
        input.simulate('keyDown', { key: 'a', stopPropagation, preventDefault });

        expect(stopPropagation).toHaveBeenCalledTimes(2);
        expect(preventDefault).toHaveBeenCalledTimes(2);
    });
});
