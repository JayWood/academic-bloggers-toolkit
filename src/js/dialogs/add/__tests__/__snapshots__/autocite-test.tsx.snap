// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoCite /> should match snapshots 1`] = `
<AutoCite
  getter={[Function]}
  kind="webpage"
  placeholder="Test"
>
  <div>
    <label
      htmlFor="citequery"
    >
      Autocite
    </label>
    <input
      id="citequery"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Test"
      type="url"
      value=""
    />
    <Button
      disabled={true}
      flat={true}
      label="Search"
      onClick={[Function]}
      type="button"
    >
      <div>
        <button
          className="btn-flat"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
        <style
          jsx={true}
        >
          
                              button {
                                  -webkit-appearance: none;
                                  -webkit-touch-callout: none;
                                  -webkit-user-select: none;
                                  -khtml-user-select: none;
                                  -moz-user-select: none;
                                  -ms-user-select: none;
                                  user-select: none;
                                  background: white;
                                  box-sizing: border-box;
                                  color: #555555;
                                  background-image: none;
                                  border: none;
                                  border-radius: 2px;
                                  box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
                                  cursor: pointer;
                                  display: inline-block;
                                  font-size: 13px;
                                  font-weight: 500;
                                  height: 36px;
                                  line-height: 36px;
                                  min-width: 64px;
                                  outline: 0;
                                  padding: 0 16px;
                                  text-align: center;
                                  text-decoration: none;
                                  text-transform: uppercase;
                                  transition: 
                      box-shadow .2s cubic-bezier(0.4, 0, 1, 1),
                      background-color .2s cubic-bezier(0.4, 0, 0.2, 1),
                      color .2s cubic-bezier(0.4, 0, 0.2, 1)
                  ;
                                  vertical-align: middle;
                                  white-space: nowrap;
                              }
                              button:hover {
                                  box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                              }
                              button:active {
                                  box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                              }
                              button:active:hover {
                                  box-shadow: 0 10px 20px rgba(0, 0, 0, .19), 0 6px 6px rgba(0, 0, 0, .23);
                                  background: rgba(0, 0, 0, 0.12);
                              }
                              button:disabled {
                                  pointer-events: none !important;
                                  color: #a0a5aa !important;
                              }
                              button.focusable:focus {
                                  outline: 1.1px auto rgba(19, 124, 189, 0.5);
                                  outline-offset: 2px;
                              }
                              button.focusable:focus:active {
                                  outline: 0;
                              }
                              .btn-flat,
                              .btn-flat.btn-primary {
                                  background: rgba(0, 0, 0, 0);
                                  box-shadow: none !important;
                              }
                              .btn-flat:hover,
                              .btn-flat.btn-primary:hover {
                                  background: rgba(158, 158, 158, .2);
                              }
                              .btn-flat:active,
                              .btn-flat.btn-primary:active {
                                  background: rgba(13, 13, 13, 0.12);
                              }
                              .btn-flat:hover:active,
                              .btn-flat.btn-primary:hover:active {
                                  background: rgba(0, 0, 0, 0.12);
                              }
                              .btn-flat.btn-primary {
                                  color: #0085ba;
                              }
                              .btn-primary {
                                  background: #0085ba;
                                  color: white;
                              }
                              .btn-primary:active:hover {
                                  background: #007aab;
                              }
                              .btn-primary:disabled {
                                  opacity: .6;
                                  box-shadow: none;
                                  color: white !important;
                              }
                              .btn-icon {
                                  border-radius: 50%;
                                  font-size: 24px;
                                  height: 32px;
                                  margin: auto 0;
                                  min-width: 32px;
                                  width: 32px;
                                  padding: 0;
                                  line-height: normal;
                              }
                              .btn-icon:hover {
                                  color: #555555;
                              }
          
                              .dashicons {
                                  line-height: 32px !important;
                                  width: 32px;
                                  height: 32px;
                              }
                              .dashicons-migrate {
                                  transform: rotateY(180deg);
                                  padding-left: 5px;
                              }
                              .dashicons-minus {
                                  font-weight: 900;
                              }
                              .dashicons-menu,
                              .dashicons-no-alt {
                                  font-size: 25px;
                              }
                              .dashicons-plus {
                                  padding-top: 2px;
                              }
                          
        </style>
      </div>
    </Button>
    <style
      jsx={true}
    >
      
                          div {
                              display: flex;
                              padding: 10px;
                              margin-bottom: 10px;
                              align-items: center;
                              background: #f5f5f5;
                              box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24), 0 -1px 0 rgba(16, 22, 26, .10);
                          }
                          #citequery {
                              margin: 0 10px;
                              flex: auto;
                              height: 28px;
                              font-size: 14px;
                          }
                      
    </style>
  </div>
</AutoCite>
`;

exports[`<AutoCite /> should match snapshots 2`] = `
<AutoCite
  getter={[Function]}
  kind="book"
  pattern="[0-9]+"
  placeholder="Test"
>
  <div>
    <label
      htmlFor="citequery"
    >
      Autocite
    </label>
    <input
      id="citequery"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern="[0-9]+"
      placeholder="Test"
      type="text"
      value=""
    />
    <Button
      disabled={true}
      flat={true}
      label="Search"
      onClick={[Function]}
      type="button"
    >
      <div>
        <button
          className="btn-flat"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
        <style
          jsx={true}
        >
          
                              button {
                                  -webkit-appearance: none;
                                  -webkit-touch-callout: none;
                                  -webkit-user-select: none;
                                  -khtml-user-select: none;
                                  -moz-user-select: none;
                                  -ms-user-select: none;
                                  user-select: none;
                                  background: white;
                                  box-sizing: border-box;
                                  color: #555555;
                                  background-image: none;
                                  border: none;
                                  border-radius: 2px;
                                  box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
                                  cursor: pointer;
                                  display: inline-block;
                                  font-size: 13px;
                                  font-weight: 500;
                                  height: 36px;
                                  line-height: 36px;
                                  min-width: 64px;
                                  outline: 0;
                                  padding: 0 16px;
                                  text-align: center;
                                  text-decoration: none;
                                  text-transform: uppercase;
                                  transition: 
                      box-shadow .2s cubic-bezier(0.4, 0, 1, 1),
                      background-color .2s cubic-bezier(0.4, 0, 0.2, 1),
                      color .2s cubic-bezier(0.4, 0, 0.2, 1)
                  ;
                                  vertical-align: middle;
                                  white-space: nowrap;
                              }
                              button:hover {
                                  box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                              }
                              button:active {
                                  box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                              }
                              button:active:hover {
                                  box-shadow: 0 10px 20px rgba(0, 0, 0, .19), 0 6px 6px rgba(0, 0, 0, .23);
                                  background: rgba(0, 0, 0, 0.12);
                              }
                              button:disabled {
                                  pointer-events: none !important;
                                  color: #a0a5aa !important;
                              }
                              button.focusable:focus {
                                  outline: 1.1px auto rgba(19, 124, 189, 0.5);
                                  outline-offset: 2px;
                              }
                              button.focusable:focus:active {
                                  outline: 0;
                              }
                              .btn-flat,
                              .btn-flat.btn-primary {
                                  background: rgba(0, 0, 0, 0);
                                  box-shadow: none !important;
                              }
                              .btn-flat:hover,
                              .btn-flat.btn-primary:hover {
                                  background: rgba(158, 158, 158, .2);
                              }
                              .btn-flat:active,
                              .btn-flat.btn-primary:active {
                                  background: rgba(13, 13, 13, 0.12);
                              }
                              .btn-flat:hover:active,
                              .btn-flat.btn-primary:hover:active {
                                  background: rgba(0, 0, 0, 0.12);
                              }
                              .btn-flat.btn-primary {
                                  color: #0085ba;
                              }
                              .btn-primary {
                                  background: #0085ba;
                                  color: white;
                              }
                              .btn-primary:active:hover {
                                  background: #007aab;
                              }
                              .btn-primary:disabled {
                                  opacity: .6;
                                  box-shadow: none;
                                  color: white !important;
                              }
                              .btn-icon {
                                  border-radius: 50%;
                                  font-size: 24px;
                                  height: 32px;
                                  margin: auto 0;
                                  min-width: 32px;
                                  width: 32px;
                                  padding: 0;
                                  line-height: normal;
                              }
                              .btn-icon:hover {
                                  color: #555555;
                              }
          
                              .dashicons {
                                  line-height: 32px !important;
                                  width: 32px;
                                  height: 32px;
                              }
                              .dashicons-migrate {
                                  transform: rotateY(180deg);
                                  padding-left: 5px;
                              }
                              .dashicons-minus {
                                  font-weight: 900;
                              }
                              .dashicons-menu,
                              .dashicons-no-alt {
                                  font-size: 25px;
                              }
                              .dashicons-plus {
                                  padding-top: 2px;
                              }
                          
        </style>
      </div>
    </Button>
    <style
      jsx={true}
    >
      
                          div {
                              display: flex;
                              padding: 10px;
                              margin-bottom: 10px;
                              align-items: center;
                              background: #f5f5f5;
                              box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24), 0 -1px 0 rgba(16, 22, 26, .10);
                          }
                          #citequery {
                              margin: 0 10px;
                              flex: auto;
                              height: 28px;
                              font-size: 14px;
                          }
                      
    </style>
  </div>
</AutoCite>
`;
